File: backend/blog-service/src/main/java/com/lily/blogservice/dto/BlogSearchRequest.java
package com.lily.blogservice.dto;

import lombok.Getter;
import lombok.Setter;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;


@Getter
@Setter
public class BlogSearchRequest {

  private String keywords;

  private SearchSort sort = SearchSort.ACCURACY;

  @Min(1) @Max(1000)
  private int page = 1;

  @Min(0) @Max(500)
  private int size = 30;

  private boolean first = false; // Marking the ranking of keyword only when 'first' is true

}

File: backend/blog-service/src/main/kotlin/com/lily/blogservice/data/BlogSearchRequest.kt
package com.lily.blogservice.data

class BlogSearchRequest(val keywords:String,
                        val sort:SearchSort=SearchSort.ACCURACY,
                        val page:Int=1,
                        val size:Int=30,
                        val first:Boolean=false)


enum class SearchSort {
    ACCURACY, RECENCY;
}